t/10-virtio_terminal.t

```diff
@@ -25,21 +27,27 @@ sub prepare_pipes ($socket_path, $write_buffer = undef) {
         mkfifo($_, 0666) or die("Cannot create fifo pipe $_");
     }
 
+    $pipe_data_written = 0;
+    $SIG{USR2} = sub { $pipe_data_written = 1 };
+
     my $pid = fork || do {
         my $running = 1;
         $SIG{USR1} = sub { $running = 0; };
         $SIG{ALRM} = sub {
             die('Timeout for pipe other side helper');
         };
         alarm ONE_MINUTE;
-
-        open(my $fd_r, "<", $pipe_in)
+        my $fd_r = IO::Handle->new();
+        my $fd_w = IO::Handle->new();
+        open($fd_r, "<", $pipe_in)
           or die "Can't open in pipe for writing $!";
-        open(my $fd_w, ">", $pipe_out)
+        open($fd_w, ">", $pipe_out)
           or die "Can't open out pipe for reading $!";
 
         syswrite($fd_w, $write_buffer) if ($write_buffer);
+        $fd_w->flush();
 
+        kill 'USR2', getppid;
```

Comment:
Wouldn't we need the parent processes' PID here?

Reply:
note the **p**pid https://perldoc.perl.org/functions/getppid

Reply:
Ah, I've missed the 2nd "p". Then it makes sense.